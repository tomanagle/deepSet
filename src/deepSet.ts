const deepSet = (arr: any[], selector: string) => {
  const uniqueSet = [
    ...new Set(
      arr.map((item: any) => {
        return item[selector]
      })
    )
  ]

  return uniqueSet.map(uniqueSelector => {
    return arr.filter((item: any) => {
      return item[selector] === uniqueSelector
    })[0]
  })
}

export default deepSet
