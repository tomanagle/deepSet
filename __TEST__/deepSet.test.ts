import deepSet from '../src/deepSet'

describe('Deep set', () => {
  it('works if true is truthy', () => {
    expect(true).toBeTruthy()
  })

  const testInput = [
    {
      name: 'Tom',
      age: 99
    },
    {
      name: 'Bob',
      age: 100
    },
    {
      name: 'Tom',
      age: 50
    }
  ]

  it('Should match snapshot', () => {
    const result = deepSet(testInput, 'name')
    expect(result).toMatchSnapshot()
  })

  it('Should have a single instead of each object', () => {
    const result = deepSet(testInput, 'name')

    const names = testInput.map(item => item.name)

    for (let i = 0; i < names.length; i += 1) {
      expect(result.filter(item => item.name === names[i]).length).toBe(1)
    }
  })
})
