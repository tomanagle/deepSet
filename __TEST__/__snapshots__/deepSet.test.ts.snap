// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deep set Should match snapshot 1`] = `
Array [
  Object {
    "age": 99,
    "name": "Tom",
  },
  Object {
    "age": 100,
    "name": "Bob",
  },
]
`;
